The job **`x86_64-mingw`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10568464988/job/29279480628) [(plain)](https://github.com/rust-lang-ci/rust/commit/17b491d34708812f3818c010216b6926e5a892d8/checks/29279480628/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  98% (48250/49234)
Updating files:  99% (48742/49234)
Updating files: 100% (49234/49234)
Updating files: 100% (49234/49234), done.
branch 'try' set up to track 'origin/try'.
Switched to a new branch 'try'
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'17b491d34708812f3818c010216b6926e5a892d8'
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
[RUSTC-TIMING] cranelift_codegen test:false 58.731
   Compiling rustc_codegen_cranelift v0.1.0 (C:\a\rust\rust\compiler\rustc_codegen_cranelift)
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-Wl,C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\list.def" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\symbols.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.00.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.01.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.02.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.03.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.04.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.05.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.06.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.07.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.08.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.09.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.10.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.11.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.12.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.13.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.14.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.rustc_codegen_cranelift.1c9dad93178a9ada-cgu.15.rcgu.o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.7k795ajmewfc5iz9uq9s0tqye.rcgu.rmeta" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\release\\deps" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_native-a679d148d60411df.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_object-948ef4e62281ca63.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libobject-f438b3f2386d6953.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcrc32fast-b5bb0d7a06f8df3f.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libmemchr-61da0b757c8589f3.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_frontend-2d64667b45a7359f.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_module-80be0c7d0b3b31de.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libanyhow-68a9151b55912af6.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_codegen-761c78f706a1c50f.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libbumpalo-6c8f495b53e87bdb.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libtarget_lexicon-1370c6482bb3b3d8.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_codegen_shared-5f6a6805afd5306b.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libregalloc2-e839fbd1ad36e233.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\liblog-f33257438d9d63b5.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libslice_group_by-c4b43037720246d0.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libhashbrown-6e465a2dc386cfd6.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libsmallvec-a6572921cdd1c4f9.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\librustc_hash-d943f4824d2824c7.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libgimli-545d97d80b8a33c0.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libindexmap-286627352e19be12.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libequivalent-46bcd7ff4e07ac14.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libhashbrown-caff8b5f3c215f13.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libahash-622f705c1b9c0fcb.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libonce_cell-dc2ab37d5f65f68b.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcfg_if-19907a9ba3f0ffc5.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libzerocopy-b262765b3c82094e.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_control-36b64cc79108a2e4.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libarbitrary-4118843f3510c500.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_bforest-f9aa41da5fde2258.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_entity-1fb678a5454b423e.rlib" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\libcranelift_bitset-7b7b92d28e7a51ac.rlib" "-Wl,-Bdynamic" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rustc_driver-ec5f8ab161f8383a.dll" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\std-9c9d9b6486848477.dll" "-Wl,-Bstatic" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-f28f71fb1ba6d9b1.rlib" "-Wl,-Bdynamic" "-lpsapi" "-lshell32" "-lole32" "-luuid" "-ladvapi32" "-lws2_32" "-lntdll" "-lkernel32" "-ladvapi32" "-lole32" "-loleaut32" "-ladvapi32" "-lcfgmgr32" "-lgdi32" "-lkernel32" "-lmsimg32" "-lopengl32" "-luser32" "-lwinspool" "-lbcrypt" "-ladvapi32" "-lwindows" "-lkernel32" "-lkernel32" "-ladvapi32" "-lntdll" "-luserenv" "-lws2_32" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\advapi32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-errorhandling-l1-1-3.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-file-fromapp-l1-1-0.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-handle-l1-1-0.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-ioring-l1-1-0.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-synch-l1-2-0.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-sysinfo-l1-2-0.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-sysinfo-l1-2-3.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-sysinfo-l1-2-4.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-sysinfo-l1-2-6.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-util-l1-1-1.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-core-wow64-l1-1-1.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\api-ms-win-security-base-l1-2-2.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\avrt.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\bcp47mrm.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\bcryptprimitives.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\clfsw32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\dbghelp.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\elscore.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\gdi32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\icu.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\imagehlp.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\kernel32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\ktmw32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\netapi32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\normaliz.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\ntdll.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\ntdllk.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\ole32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\oleacc.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\oleaut32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\psapi.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\rtworkq.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\txfw32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\user32.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\usp10.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\version.dll_imports_indirect.lib" "C:\\a\\_temp\\msys64\\tmp\\rustc8Vjh1O\\wofutil.dll_imports_indirect.lib" "-lgcc_s" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmingwex" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-o" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\rustc_codegen_cranelift-12242d8aea75ff51.dll" "-shared" "-Wl,--out-implib=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-codegen\\x86_64-pc-windows-gnu\\release\\deps\\librustc_codegen_cranelift-12242d8aea75ff51.dll.a" "-Wl,-O1" "-nodefaultlibs" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-gnu\\stage0-sysroot\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:/a/rust/rust/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lwindows: No such file or directory
          collect2.exe: error: ld returned 1 exit status
          

[RUSTC-TIMING] rustc_codegen_cranelift test:false 14.899
```

</details>

--- a/compiler/rustc_data_structures/Cargo.toml
+++ b/compiler/rustc_data_structures/Cargo.toml
@@ -22,7 +22,7 @@ rustc_index = { path = "../rustc_index", package = "rustc_index" }
 rustc_macros = { path = "../rustc_macros" }
 rustc_serialize = { path = "../rustc_serialize" }
 smallvec = { version = "1.8.1", features = ["const_generics", "union", "may_dangle"] }
-stacker = "0.1.15"
+stacker = "0.1.17"
 tempfile = "3.2"
 thin-vec = "0.2.12"
 tracing = "0.1"

